test/e2e/workflow/deployment_test.go

```diff
@@ -454,6 +455,124 @@ var _ = Describe("NetworkAddonsConfig", func() {
+					HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal),
+					HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal),
+					HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal)))
+			})
+
+			It("Should take the `part-of`, `version` and `component` labels value and apply it to the deployed daemonset", func() {
+				Eventually(func() *v1.DaemonSet {
+					err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: OvsComponent.DaemonSets[0], Namespace: components.Namespace}, ovsDaemonSet)
+					Expect(err).ToNot(HaveOccurred(), "should succeed getting the ovs daemonSet")
+					return ovsDaemonSet
+				}, 2*time.Minute, time.Minute).Should(SatisfyAll(
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDaemonsetLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDaemonsetTemplateLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal))), "label and template label values of the daemonset should be contain the values on NetworkAddonsConfig")
+			})
+
+			It("Should take the `part-of`, `version` and `component` labels value and apply it to the deployed deployments", func() {
+				Eventually(func() *v1.Deployment {
+					err := framework.Global.Client.Get(context.TODO(), types.NamespacedName{Name: KubeMacPoolComponent.Deployments[0], Namespace: components.Namespace}, kubemacpoolDeployment)
+					Expect(err).ToNot(HaveOccurred(), "should succeed getting the kubemacpool deployment")
+					return kubemacpoolDeployment
+				}, 2*time.Minute, time.Minute).Should(SatisfyAll(
+					WithTransform(getDeploymentLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDeploymentLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDeploymentLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal)),
+					WithTransform(getDeploymentTemplateLabels, HaveKeyWithValue(names.PART_OF_LABEL_KEY, partOfLabelCRVal)),
+					WithTransform(getDeploymentTemplateLabels, HaveKeyWithValue(names.VERSION_LABEL_KEY, versionLabelCRVal)),
+					WithTransform(getDeploymentTemplateLabels, HaveKeyWithValue(names.COMPONENT_LABEL_KEY, componentLabelCRlVal))), "label and template label values of the deployment should be contain the values on NetworkAddonsConfig")
+			})
+		})
+
+		Context("When NetworkAddonsConfig is labeled with `managed-by` label", func() {
+			const (
+				managedByLabelCRVal        = "managed-by-cr-value"
+				managedByLabelComponentVal = names.MANAGED_BY_LABEL_DEFAULT_VALUE
```

Comment:
better to have this line first as this is the default value

Reply:
done